### ISSUE TITLE
HardLib is causing serious lag spikes

### ISSUE THREAD

**Issue by Xonok** [Draco18s/ReasonableRealism#48]
I'm building a modpack that I keep highly optimized. Recently I found that something is causing huge lag spikes, sometimes of several seconds.
Spark has concluded that around 87% of tick length during lag spikes is caused by this one function in HardLib.
What do I need to do to turn this part off?
I'm using this version of HardLib(https://github.com/Draco18s/ReasonableRealism/issues/36), but the lag was also there with the latest 1.12 from curseforge. 
![2022-01-06 05_29_02-Window](https://user-images.githubusercontent.com/5237777/148323791-4c2e2cb6-0140-4e6d-8afc-1e2a7855e508.png)
![2022-01-06 05_31_20-Profile at 05_12AM 1_6_2022 — Mozilla Firefox](https://user-images.githubusercontent.com/5237777/148323956-b160f255-afdf-4057-96e1-5d1b3b7b0980.png)

**Comment by Draco18s** [comment]
Yeah, that'd be this method:
https://github.com/Draco18s/ReasonableRealism/blob/1.12.1/src/main/java/com/draco18s/hardlib/api/recipes/RecipeToolMold.java#L117

If you don't care about the Tool Mold feature of Expanded Industry, you could create a recipe data pack and override all of the mold_xxx recipes:
https://github.com/Draco18s/ReasonableRealism/tree/1.12.1/src/main/resources/assets/expindustry/recipes

I'll see about trying to optimize that method, but I'm not sure how much I can do.

**Comment by Xonok** [comment]
Yeah, don't really care about the molds. It seems turning them off through crafttweaker works.
